engine/src/cmd/unit_csv.cpp

```diff
@@ -1088,162 +850,142 @@ void Unit::LoadRow( CSVRow &row, string modification, string *netxml )
-        std::string iffval = OPTIM_GET( row, table, Radar_Color );
-        if ((iffval.empty()) || (iffval == "FALSE") || (iffval == "0"))
-        {
-            computer.radar.capability = Computer::RADARLIM::Capability::IFF_NONE;
-        }
-        else if ((iffval == "TRUE") || (iffval == "1"))
-        {
-            computer.radar.capability
-                = Computer::RADARLIM::Capability::IFF_SPHERE
+        computer.radar.capability = Computer::RADARLIM::Capability::IFF_NONE;
+    } else if ((iffval == "TRUE") || (iffval == "1")) {
+        computer.radar.capability = Computer::RADARLIM::Capability::IFF_SPHERE
                 | Computer::RADARLIM::Capability::IFF_FRIEND_FOE;
-        }
-        else if (iffval == "THREAT")
-        {
-            computer.radar.capability
-                = Computer::RADARLIM::Capability::IFF_SPHERE
+    } else if (iffval == "THREAT") {
+        computer.radar.capability = Computer::RADARLIM::Capability::IFF_SPHERE
                 | Computer::RADARLIM::Capability::IFF_FRIEND_FOE
                 | Computer::RADARLIM::Capability::IFF_THREAT_ASSESSMENT;
-        }
-        else if (iffval == "BUBBLE_THREAT")
-        {
-            computer.radar.capability
-                = Computer::RADARLIM::Capability::IFF_BUBBLE
+    } else if (iffval == "BUBBLE_THREAT") {
+        computer.radar.capability = Computer::RADARLIM::Capability::IFF_BUBBLE
                 | Computer::RADARLIM::Capability::IFF_FRIEND_FOE
                 | Computer::RADARLIM::Capability::IFF_OBJECT_RECOGNITION
                 | Computer::RADARLIM::Capability::IFF_THREAT_ASSESSMENT;
-        }
-        else if (iffval == "PLANE")
-        {
-            computer.radar.capability
-                = Computer::RADARLIM::Capability::IFF_PLANE
+    } else if (iffval == "PLANE") {
+        computer.radar.capability = Computer::RADARLIM::Capability::IFF_PLANE
                 | Computer::RADARLIM::Capability::IFF_FRIEND_FOE;
-        }
-        else if (iffval == "PLANE_THREAT")
-        {
-            computer.radar.capability
+    } else if (iffval == "PLANE_THREAT") {
+        computer.radar.capability
                 = Computer::RADARLIM::Capability::IFF_PLANE
                 | Computer::RADARLIM::Capability::IFF_FRIEND_FOE
                 | Computer::RADARLIM::Capability::IFF_OBJECT_RECOGNITION
                 | Computer::RADARLIM::Capability::IFF_THREAT_ASSESSMENT;
-        }
-        else
-        {
-            unsigned int value = stoi(iffval, 0);
-            if (value == 0)
-            {
-                // Unknown value
-                assert(false);
-                computer.radar.capability = Computer::RADARLIM::Capability::IFF_NONE;
-            }
-            else
-            {
-                computer.radar.capability = value;
-            }
+    } else {
+        unsigned int value = stoi(iffval, 0);
+        if (value == 0) {
+            // Unknown value
+            assert(false);
```

Comment:
assert should have a good message with it for the logs so that we can know why it broke...or was this just for your testing and it slipped through?

Reply:
I was actually confused by it, as I thought it was mine, but it's existing code. I think I'll just delete it, as it's clearly not intended to be there. I think someone else added it for exactly the same reason - to catch some edge case and accidentally left it there.